### PR TITLE
Update to match Hixie's changed bash script

### PR DESCRIPTION
**Pull request by TheBosZ**
https://github.com/Hixie/flutter/blob/flutter-analyze-PENDING/bin/flutter

### CODE REVIEW

File: bin/flutter.bat [hunk 1]
```diff
@@ -8,38 +8,36 @@ FOR %%i IN ("%~dp0..") DO SET "flutter_root=%%~fi" REM Get the parent directory
...
+CD "%flutter_root%"
 CALL %dart% --snapshot="%snapshot_path%" --package-root="%flutter_tools_dir%\packages" "%script_path%"
-ECHO !revision! > "%stamp_path%"
+<nul SET /p=%revision%> "%stamp_path%"
+goto :eof
 
 :after_snapshot
 
-CALL %dart% "%snapshot_path%" %*
+CALL %dart% --checked "%snapshot_path%" %*
```
Comment (Hixie) [inline bin/flutter.bat]: oops, don't include this one. That was just for me to debug the changes I was making to the flutter tool. :-)

File: bin/flutter.bat [hunk 2]
```diff
@@ -8,38 +8,36 @@ FOR %%i IN ("%~dp0..") DO SET "flutter_root=%%~fi" REM Get the parent directory
...
 GOTO :after_snapshot
 
 :do_snapshot
-
+CD "%flutter_tools_dir%"
+CALL pub.bat get
+CD "%flutter_root%"
 CALL %dart% --snapshot="%snapshot_path%" --package-root="%flutter_tools_dir%\packages" "%script_path%"
-ECHO !revision! > "%stamp_path%"
+<nul SET /p=%revision%> "%stamp_path%"
```
Comment (Hixie) [inline bin/flutter.bat]: what dark magic is this!

**Comment by TheBosZ** [comment]
I was wondering about that, but thought you knew what you were doing!

**Comment by Hixie** [comment]
LGTM, I'll check in the bash equivalent as part of my `dart analyze` patch.

File: bin/flutter.bat (hunk 2 above)
Comment (TheBosZ) [inline bin/flutter.bat]: The old way leaves a line feed. This way, it takes advantage of SET /p refusing whitespace: http://stackoverflow.com/questions/7105433/windows-batch-echo-without-new-line

**Comment by TheBosZ** [comment]
:+1:

**Pull request merged**